src/Message/AbstractRequest.php

```diff
@@ -232,47 +208,42 @@ public function setUsername($username)
     }
 
     /**
-     * @param Response $response
+     * @param ResponseInterface $response
      *
      * @return array
-     *
-     * @throws RuntimeException
      */
-    protected function getResponseBody(Response $response)
+    protected function getResponseBody(ResponseInterface $response)
     {
-        return empty($response->getBody(true)) ? [] : $response->json();
+        return \json_decode($response->getBody()->getContents(), true);
     }
 
     /**
      * @param string $method
      * @param string $url
      * @param mixed  $data
      *
-     * @return Response
-     *
-     * @throws RequestException
-     * @throws InvalidArgumentException
+     * @return ResponseInterface
      */
     protected function sendRequest($method, $url, $data)
     {
         $options = (new AuthenticationRequestOptionProvider())->getOptions($this);
 
-        if (RequestInterface::GET === $method) {
-            return $this->httpClient->createRequest(
+        if ('GET' === $method) {
+            return $this->httpClient->request(
```

Comment:
May throw `\Omnipay\Common\Http\Exception\NetworkException` and `\Omnipay\Common\Http\Exception\RequestException`